homeassistant/components/enphase_envoy/config_flow.py

```diff
@@ -213,3 +213,69 @@ async def async_step_user(
             description_placeholders=description_placeholders,
             errors=errors,
         )
+
+    async def async_step_reconfigure(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Add reconfigure step to allow to manually reconfigure a config entry."""
+        errors: dict[str, str] = {}
+        description_placeholders: dict[str, str] = {}
+
+        entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        assert entry
+
+        host: Any = (user_input or entry.data).get(CONF_HOST)
+        username: Any = (user_input or entry.data).get(CONF_USERNAME)
+        password: Any = (user_input or entry.data).get(CONF_PASSWORD)
+
+        suggested_values = {
+            CONF_HOST: host,
+            CONF_USERNAME: username,
+            CONF_PASSWORD: password,
+        }
+
+        if user_input is not None:
+            try:
+                envoy = await validate_input(
+                    self.hass,
+                    host,
+                    username,
+                    password,
+                )
+            except INVALID_AUTH_ERRORS as e:
+                errors["base"] = "invalid_auth"
+                description_placeholders = {"reason": str(e)}
+            except EnvoyError as e:
+                errors["base"] = "cannot_connect"
+                description_placeholders = {"reason": str(e)}
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                if not self.unique_id:
+                    await self.async_set_unique_id(envoy.serial_number)
+
+                if self.unique_id:
+                    # If envoy exists in configuration update fields and exit
+                    self._abort_if_unique_id_configured(
+                        {
+                            CONF_HOST: host,
+                            CONF_USERNAME: username,
+                            CONF_PASSWORD: password,
+                        },
+                        error="reconfigure_successful",
+                    )
```

Comment:
Shouldn't we check if the same unique_id is used?

Reply:
It is getting the serialnumber from the Envoy and uses that for the unique_id. If it would get a different serialnumber when they changed the IP, it would mean there's another Envoy and it will update that Envoy instead off the one they started from. There's probably people out there that would consider it a feature, but I'll add a test for it to catch that and signal that it's not the same Envoy that would get changed. Long story for a short yes.

Reply:
> it will update that Envoy instead off the one they started from

I did not expect that to happen.

Also, the code says `if not self.unique_id`: Is there a way there are entries without unique_ids?

Reply:
> Also, the code says if not self.unique_id: Is there a way there are entries without unique_ids?
No, but self.unique_id is the configflow unique_id. It never got stored there, so I'm changing code to do that at first entry so it remembers when coming back around after user inpyt.
> I did not expect that to happen.
Form let's user change ip, for the case some network cahnged happened that are not picked up automatically through zeroconf. Validate input will probe at that ip and get serial back, if its a different envoy and existing in HA, that one will update. That's the reason why your remark is valuable, so we only change serial that we started with.

So with that unique_id now in flow that should prevent it.

Reply:
implemented